jobs/integration/utils.py

```diff
@@ -496,37 +501,40 @@ async def validate_storage_class(model, sc_name, test_name):
         )
 
 
-async def _debug_ceph_storage(control_plane, pods):
-    log.error("Gathering Ceph logs from Cluster")
+async def _debug_storage_class(debug_open, test_name, sc_name, provisioner, model):
+    class _Call:
+        def __init__(self, *args, **kwds):
+            self.args, self.kwds = args, kwds
 
-    base_command = "/snap/bin/kubectl --kubeconfig /root/.kube/config"
-    resources = [
-        ("Nodes status", f"{base_command} describe nodes"),
-        ("PVCs status", f"{base_command} describe pvc"),
-        ("Storage Classes status", f"{base_command} describe sc -A"),
-        *(
-            (f"Pod {pod_name} status", f"{base_command} describe pod {pod_name}")
-            for pod_name in pods
-        ),
-    ]
-
-    for msg, cmd in resources:
-        output = await juju_run(control_plane, cmd)
-        log.error(f"{msg}: {output.output}")
-
-    cmd = (
-        base_command
-        + " get pods -l app=csi-rbdplugin-provisioner -o jsonpath='{.items[*].metadata.name}'"
-    )
-    output = await juju_run(control_plane, cmd)
-    rbd_prov_pods = output.output.split(" ")
-    log.error("RBD Provisioner status: ")
-    for pod in rbd_prov_pods:
-        cmd = f"{base_command} logs {pod} > logs.txt"
-        output = await juju_run(control_plane, cmd)
-        cmd = "cat logs.txt"
-        output = await juju_run(control_plane, cmd)
-        log.error(output.output)
+    pods = [f"{sc_name}-write-test", f"{sc_name}-read-test"]
+    namespace = "default"
+    log.info(f"Gathering {test_name} storage logs from cluster")
+    provisioner_pods = []
+    if provisioner:
+        result = await kubectl(
+            model,
+            "get",
+            "pods",
+            A=True,
+            l=f"app={provisioner}",
+            o=r"""jsonpath='{range .items[*]}{@..metadata.namespace}{" "}{@..metadata.name}{"\n"}{end}'""",
+        )
```

Comment:
this `jsonpath` formats the pods as `<pod-namespace> <pod-name` per line of output

using that, we can parse the output lines and look up the logs of the provisioner further down